skyportal/models.py

```diff
@@ -134,10 +142,17 @@ class Source(Base):
     tns_name = sa.Column(sa.Unicode, nullable=True)
 
     groups = relationship('Group', secondary='group_sources')
+    candidate_groups = relationship("Group", secondary="group_candidates")
     comments = relationship('Comment', back_populates='source',
+                            primaryjoin="comments.c.source_id == sources.c.id",
                             cascade='save-update, merge, refresh-expire, expunge',
                             passive_deletes=True,
                             order_by="Comment.created_at")
+    candidate_comments = relationship("Comment", back_populates="candidate",
+                                      primaryjoin="comments.c.candidate_id == sources.c.id",
+                                      cascade="save-update, merge, refresh-expire, expunge",
```

Comment:
Any reason why we shouldn't use `cascade="all"`, which [translates to](https://docs.sqlalchemy.org/en/13/orm/cascades.html) `save-update, merge, refresh-expire, expunge, delete`?

Reply:
We specifically don't want to include delete here (that's handled in the foreign key specification, not in the relationship)